### ISSUE TITLE
Plugins not working properly when using compileMdx

### ISSUE THREAD

**Issue by billgun** [hashicorp/next-mdx-remote#431]
As the title says, or i might be missing something but i cant find anywhere in the docs

As the title says
```ts
export async function getMdx({ source }: MdxProps) {
  const { content, frontmatter } = await compileMDX<any>({
    source,
    components,
    options: {
      parseFrontmatter: true,
      mdxOptions: { remarkPlugins: [remarkToc, remarkGfm], format: 'mdx' },
    },
  });
  return {
    content,
    frontmatter,
  };
}

```

```ts
async function getDocFromParams({ params }: DocPageProps) {
  const slug = params.slug?.join('/') || 'index';
  const filePath = path.join(process.cwd(), docDir, `${slug}.mdx`);

  if (fs.existsSync(filePath)) {
    const source = fs.readFileSync(filePath, 'utf8');
    const { content, frontmatter } = await getMdx({ source });

    return {
      meta: frontmatter,
      slug,
      content,
    };
  } else {
    return null;
  }
```
My mdx is not rendering properly with plugins used.
![image-9.png](https://github.com/hashicorp/next-mdx-remote/assets/22515292/5fd1454e-0a92-4488-83dd-c337917039ad)

**Comment by BillGun** [comment]
Anothere reference i have tried as well [url](https://www.answeroverflow.com/m/1167726380998463519#solution-1168624164404478123)

**Comment by turtle601** [comment]
I have a same problem too

**Comment by talatkuyuk** [comment]
**Can you share a minimum complete app** in order to understand problem ? 
_What is your mdx content? what is your main rendered component? etc..._

**Comment by BillGun** [comment]
Apparently my markdown is wrong because I just copied it, I am sorry :/
The markdown for the table is all in one line, but when I fixed it I ran into another error.
![image](https://github.com/hashicorp/next-mdx-remote/assets/22515292/f29ef7c0-8b4f-4cf9-ab24-55522f47693b)
But this can be fixed by downgrading remark-gfm to ^3.0.0 instead of ^4.0.0
I am not sure if this is next-mdx-remote issues or remark-gfm? [url](https://github.com/remarkjs/remark-gfm/issues/57)

**Comment by talatkuyuk** [comment]
The issue is related with `next-mdx-remote` not `remark-gfm`. The `remark-gfm` ^3.0.0 and ^4.0.0 uses different version of `unified`. And current `next-mdx-remote`is not inline with latest version the `remark-gfm` ^4.0.0

**Comment by firmart** [comment]
@talatkuyuk Thanks. I downgraded `remark-gfm` to `3.0.1`. Works fine.

**Comment by karlhorky** [comment]
I can confirm that `next-mdx-remote@0.0.0-canary-20240321205249` (canary version from [the v5 PR](https://github.com/hashicorp/next-mdx-remote/pull/448#issuecomment-1995658726)) has resolved the incompatibility with `remark-gfm@4.0.0` 🎉

![Screenshot 2024-04-06 at 16 35 35](https://github.com/hashicorp/next-mdx-remote/assets/1935696/5b61858b-a6e6-4d66-86ba-a89dff62394c)

**Comment by dstaley** [comment]
@karlhorky thanks for confirming!
